apps/server/src/modules/news/controller/news.controller.ts

```diff
@@ -26,22 +31,41 @@ export class NewsController {
 		return dto;
 	}
 
+	/**
+	 * Responds with all news for a user.
+	 */
 	@Get()
 	async findAll(
 		@CurrentUser() currentUser: ICurrentUser,
 		@Query() scope: NewsFilterQuery,
 		@Query() pagination: PaginationQuery
 	): Promise<PaginationResponse<NewsResponse[]>> {
-		const [newsList, count] = await this.newsUc.findAllForUser(
-			currentUser.userId,
-			NewsMapper.mapNewsScopeToDomain(scope),
-			pagination
-		);
-		const dtoList = newsList.map((news) => NewsMapper.mapToResponse(news));
-		return new PaginationResponse(dtoList, count);
+		const newsResponse = await this.find(currentUser, scope, { pagination });
+		return newsResponse;
+	}
+
+	/**
+	 * Responds with news of a given team for a user.
+	 */
+	@Get('/team/:teamId')
+	async findAllForTeam(
```

Comment:
The team ID is a filter criteria here. Thus I would prefer the teamId to be a query param. 
Path params usually mean you are looking for a specific resource. So just reading the URL I would guess the response type of
news/team/:id   -> team (the team resource that is part of news with ID :id)
new?teamId=:id   -> news (the news with a teamID=:id)

Reply:
We added the controller route because it is used in a separate client view to list team news. Both cases are possible. We decided to a separate controller method as it simplifies the clients request and we know that team news are in use by a client which were now added to e2e tests to ensure this works.

Reply:
But in REST news/team usually means you get the team of the news.
team/:id/news would mean you get the news of the team.

I guess, both my suggestions make the server code more complicated. But from the outside view it would make more sense.

Reply:
the question in general is how to handle teams in news...
I added a new controller implementing `team/:teamId/news` now